### ISSUE TITLE
Using local registry with nss-myhostname

### ISSUE THREAD

**Issue by remche** [okteto/okteto#2416]
**Describe the bug**
Using a local registry with [nss-myhostname](https://man7.org/linux/man-pages/man8/nss-myhostname.8.html) name (i.e. registry.localhost) does not work with okteto 2.

**To Reproduce**
Use `k3d-registry.localhost:12345/image` as image name for build.service.image.
Use `okteto up` will thrown an error, trying to resolve `k3d-registry.localhost` (see logs).
Note that `okteto build` works fine.

**Desktop (please complete the following information):**

- OS: Arch Linux
- Version okteto 2.0.1

**Additional context**

Workaround is to use
`localhost:12345/image`  for build.service.image.
and
`k3d-registry.localhost:12345/image`  for dev.service.image.

```
Error checking image at registry k3d-registry.localhost:12345/xxx: error getting image tag digest: Get "https://k3d-registry.localhost:12345/v2/": dial tcp: lookup k3d-registry.localhost on xx.xx.xx.xx:53: no such host; Get "http://k3d-registry.localhost:12345/v2/": dial tcp: lookup k3d-registry.localhost on xx.xx.xx.xx:53: no such host
```

**Comment by pchico83** [comment]
@remche thanks for reporting this, looks like a very interesting use case.
Could you rephrase the workaround, seems like you said to use two different values at `build.service.image`.

Als, could you share your okteto manifest?

**Comment by remche** [comment]
> @remche thanks for reporting this, looks like a very interesting use case. Could you rephrase the workaround, seems like you said to use two different values at `build.service.image`.

oops, sorry, I just fixed that in issue description.
* for build image we need to use `localhost` as registry because of the `Error checking image...` bug ;
* for dev image we use `k3d-registry.localhost` because we use [K3d managed registry](https://k3d.io/v5.2.0/usage/registries/#using-k3d-managed-registries).

Using K3d + Okteto is a great combo for on premise dev environment !

> Als, could you share your okteto manifest?

v2 manifest :
```
build:
  chronosheep:
    image: k3d-registry.localhost:12345/chronosheep
    dockerfile: ./docker/Dockerfile
    target: dev
deploy:
  compose:
    - ./okteto-stack.yml
dev:
  chronosheep:
    autocreate: true
    image: k3d-registry.localhost:12345/chronosheep
    command: |
      /usr/local/bin/docker-entrypoint.sh unitd --no-daemon --control unix:/var/run/control.unit.sock
    sync:
      - ./django-application:/app
    forward:
      - 8000:80
      - 5432:db:5432
      - 9000:minio:9000
      - 9001:minio:9001
```

FYI, this is the v1 manifest which did not have the issue : 

```name: chronosheep
image:
  name: k3d-registry.localhost:12345/chronosheep:dev
  dockerfile: ./docker/Dockerfile
  target: dev
command: |
  /usr/local/bin/docker-entrypoint.sh unitd --no-daemon --control unix:/var/run/control.unit.sock
sync:
  - ./django-application:/app
interface: 0.0.0.0
forward:
  - 8000:80
  - 5432:db:5432
  - 9000:minio:9000
  - 9001:minio:9001
autocreate: true
```

**Comment by pchico83** [comment]
@remche awesome, thanks! We will be working on the next month planning

**Comment by jmacelroy** [comment]
@pchico83 Should this be part of the 2.2 release?

**Comment by pchico83** [comment]
@jmacelroy 2.2 is around the corner, I would add it to the 2.3 milestone

**Comment by teresaromero** [comment]
Using k3d for local development with local registry

While investigating this issue, couple things regarding setting up the local cluster and the local registry came up:

Case : create registry with name `registry.localhost` and images tagged with `k3d-registry.localhost:50418`
- No mapped hosts
    - Allowed to access via curl via API both `http://k3d-registry.localhost:<port>/v2` and `http://registry.localhost:<port>/v2`
    - Okteto build -> triggers error Get "https://k3d-registry.localhost:50418/v2/": Failed to lookup host: k3d-registry.localhost
    - Okteto up -> Error checking image at registry k3d-registry.localhost:50418/api:dev: error getting image tag digest: Get "https://k3d-registry.localhost:50418/v2/": dial tcp: lookup k3d-registry.localhost on 80.58.61.250:53: no such host; Get "http://k3d-registry.localhost:50418/v2/": dial tcp: lookup k3d-registry.localhost on 80.58.61.250:53: no such host
- Having mapped `127.0.0.1 registry.localhost`
    - Okteto build -> triggers error Get "https://k3d-registry.localhost:50418/v2/": Failed to lookup host: k3d-registry.localhost
    - Okteto up -> Error checking image at registry k3d-registry.localhost:50418/api:dev: error getting image tag digest: Get "https://k3d-registry.localhost:50418/v2/": dial tcp: lookup k3d-registry.localhost on 80.58.61.250:53: no such host; Get "http://k3d-registry.localhost:50418/v2/": dial tcp: lookup k3d-registry.localhost on 80.58.61.250:53: no such host
- Having mapped `127.0.0.1 k3d-registry.localhost`
    - Okteto build -> pushes the image correctly
    - Okteto up -> new error hit Couldn't activate your development container Error: ImagePullBackOff
    - Okteto is able to call api, but local cluster, when trying to resolve the image for the container, is unable to access, if you make a describe for the pods that have this error: Failed to pull image "k3d-registry.localhost:50418/……”..: rpc error: code = Unknown desc = failed to pull and unpack image "k3d-registry.localhost:50418/….”: failed to resolve reference "k3d-registry.localhost:50418/.”….: failed to do request: Head "https://k3d-registry.localhost:50418/v2/…”.: dial tcp 127.0.0.1:50418: connect: connection refused -> although if you try to access from your local machine via curl to this url (http) the manifest is there. So the cluster is not able to access the local registry.
- Changing the image tag to `registry.localhost:50418/<image name>`
    - Okteto build ok
    - Okteto deploy ok
    - Okteto up ok

@remche could you try again using this settings for the mapped hosts, images names and local cluster/registry setup ? 
- Create the registry with <name> and map 127.0.0.1 to this <name>
- Tag the images using the same <name> as registry, although internally k3d-<name> will work if both `registry.localhost` and `k3d-registry.localhost` are mapped to 127.0.0.1, in order to work when pulling the images inside the cluster, the tag should have the name as declared when creating registry and cluster.

Some k3d commands i used and worked to create registry and clusters:

````
# create first the registry then the cluster
# image tags with k3d-registry.localhost:50418

❯ k3d registry create registry.localhost --port 50418
❯ k3d cluster create --registry-use k3d-registry.localhost:50418
````

````
# create cluster and registry with the same command

# image tags with k3d-registry.localhost:<port>

# option 1:
❯ k3d cluster create --registry-create k3d-registry.localhost -> will choose random port and `k3d-default` as cluster name

# option 2
❯ k3d cluster create --registry-create k3d-registry.localhost:50418 -> will set port and `k3d-default` as cluster name and registry will use the port specified

# option 3
❯ k3d cluster create <name> --registry-create k3d-registry.localhost:50418 -> will set registry port to specified and `k3d-<name>` as cluster name
````

**Comment by remche** [comment]
@teresaromero thanks for you work on this.
I confirm that it works by manually setting the name in `/etc/hosts` but the topic of the issue is to use nss-myhostname to keeps the thing automatic. ;)

After digging it seems that behaviour comes from google/go-containerregistry packages :

```
/crane manifest k3d-registry.localhost:12345/chronosheep:dev
Error: fetching manifest k3d-registry.localhost:12345/chronosheep:dev: Get "https://k3d-registry.localhost:12345/v2/": dial tcp: lookup k3d-registry.localhost on 195.83.24.30:53: no such host; Get "http://k3d-registry.localhost:12345/v2/": dial tcp: lookup k3d-registry.localhost on 195.83.24.30:53: no such host
```

**Comment by teresaromero** [comment]
@remche i can't reproduce it :/ can you give some more detail on how the nss-myhostname is interfering with the package? 
thnks

**Comment by remche** [comment]
@teresaromero with nsss-myhostname, any *.localhost should resolve to 127.0.0.1.
`crane` try to resolve with dns instead.

it seems that go net package handles this though : 
https://go-review.googlesource.com/c/go/+/33550/
